crates/script/src/multi_sequence.rs

```diff
@@ -0,0 +1,152 @@
+use super::sequence::{sig_to_file_name, ScriptSequence, SensitiveScriptSequence, DRY_RUN_DIR};
+use eyre::{ContextCompat, Result, WrapErr};
+use foundry_cli::utils::now;
+use foundry_common::fs;
+use foundry_compilers::ArtifactId;
+use foundry_config::Config;
+use serde::{Deserialize, Serialize};
+use std::{
+    io::{BufWriter, Write},
+    path::PathBuf,
+};
+
+/// Holds the sequences of multiple chain deployments.
+#[derive(Clone, Default, Serialize, Deserialize)]
+pub struct MultiChainSequence {
+    pub deployments: Vec<ScriptSequence>,
+    #[serde(skip)]
+    pub path: PathBuf,
+    #[serde(skip)]
+    pub sensitive_path: PathBuf,
+    pub timestamp: u64,
+}
+
+/// Sensitive values from script sequences.
+#[derive(Clone, Default, Serialize, Deserialize)]
+pub struct SensitiveMultiChainSequence {
+    pub deployments: Vec<SensitiveScriptSequence>,
+}
+
+fn to_sensitive(sequence: &mut MultiChainSequence) -> SensitiveMultiChainSequence {
```

Comment:
Does this need `&mut`? Also I'd prefer this be like `SensitiveMultiChainSequence::from_multi_sequence` or a From impl